solution/index.js

```diff
@@ -0,0 +1,263 @@
+const localSave=localStorage.getItem("tasks")
+let existTaskText
+let data
+if(localSave!=undefined){
+   data=JSON.parse(localSave)
+   generateExistsTasks()
+}
+else{
+    data={
+        "todo":[],
+        "in-progress" :[],
+        "done":[]
+    }
+}
+
+
+//function that adds a task to the addTasks section
+function addNewTask({target}){
+    const section=target.closest("section")
+    const newTask=section.querySelector("input").value
+    if(newTask===""){
+        alert("cannot add empty task")
+    }
+    else{
+        const sectionUl=section.querySelector("ul")
+        const newTaskEl=createLiElement(newTask)
+        sectionUl.prepend(newTaskEl)
+        section.querySelector("input").value=""
+        addTaskData(target,newTask)
+        saveLocalData()
+    }
+
+}
+
+//add the new data to the data variable
+function addTaskData(target,newTask){
+    if(target.id==="submit-add-to-do"){
+        data.todo.push(newTask)
+    }
+    if(target.id==="submit-add-in-progress"){
+        data["in-progress"].push(newTask)
+    }
+    if(target.id==="submit-add-done"){
+        data.done.push(newTask)
+    }
+}
+
+//save every change to thr local data
+function saveLocalData(){
+    const save=JSON.stringify(data)
+    localStorage.setItem("tasks",save)
+}
+
+
+//enable the double click function
+function editTask(targetTask){
+   existTaskText=targetTask.target.innerHTML
+   targetTask.target.contentEditable = true
+}
+
+
+
+//checks when the finished to change a task  and if he did not changed it to empty task
+function outOfEditTask(targetTask){
+    targetTask.target.contentEditable = false
+    const targetTasksSection=targetTask.target.closest("section")
+    const targetTaskSubmit=targetTasksSection.querySelector("button")
+    const innerText=targetTask.target.innerText
+    if(innerText===""){
+        alert("cannot edit to empty task")
+        targetTask.target.innerText=existTaskText
+    }
+    else{
+        const targetTaskSubmitId=targetTaskSubmit.id
+        let dataKey
+        if(targetTaskSubmitId==="submit-add-to-do")dataKey="todo"
+        if(targetTaskSubmitId==="submit-add-in-progress")dataKey="in-progress"
+        if(targetTaskSubmitId==="submit-add-done")dataKey="done"
+        updateTask(dataKey,targetTask.target.innerText)
+    }
+}
+
+
+
+//update the changed task
+function updateTask(dataKey,newTaskText){
+    arrayOfData=data[dataKey]
+          for(let i=0;i<arrayOfData.length;i++){
+              if(arrayOfData[i]===existTaskText){
+                  data[dataKey][i]=newTaskText
+                  saveLocalData()
+                  return
+              }
+          }
+}
+
+
+let taskToMove
```

Comment:
Do not use global variables. It's a sure recipe for bugs. Find a different way to pass the hovered element to the keyboard event handler.